src/routes/backup/BackupDetail.js

```diff
@@ -22,15 +22,15 @@ function Backup({ backup, volume, setting, backingImage, loading, location, disp
   const defaultReplicaCountSetting = settings.find(s => s.id === 'default-replica-count')
   const defaultNumberOfReplicas = defaultReplicaCountSetting !== undefined ? parseInt(defaultReplicaCountSetting.value, 10) : 3
 
-  const { volumeName } = queryString.parse(location.search)
-  const currentBackUp = backupVolumes.find((item) => item.volumeName === volumeName) || {}
+  const { keyword } = queryString.parse(location.search)
+  const currentBackUp = backupVolumes.find(bkVol => bkVol.name === keyword) || {}
   const v1DataEngineEnabledSetting = settings.find(s => s.id === 'v1-data-engine')
   const v2DataEngineEnabledSetting = settings.find(s => s.id === 'v2-data-engine')
   const v1DataEngineEnabled = v1DataEngineEnabledSetting?.value === 'true'
   const v2DataEngineEnabled = v2DataEngineEnabledSetting?.value === 'true'
 
-  const backups = volumeName ? backupData.filter(bk => bk.volumeName === volumeName) : backupData
-  sortBackups(backupData)
+  const backups = currentBackUp ? backupData.filter(bk => bk.volumeName === currentBackUp.volumeName) : backupData
```

Comment:
We might need to add the `backuptarget` filter to the URL as well?
The scenarios are as follows:
- Create a volume with the backup target `default` and create a backup. (e.g. 1217-test)
- Change the backup target to `cifs` and create another backup.
- On the backup page, both backups appear with their respective targets (expected).
![Screenshot 2024-12-20 at 10 40 38 AM (2)](https://github.com/user-attachments/assets/9c615b8d-d746-438d-8d18-954f76fd2c66)
- Clicking the volume with the `default` target shows `cifs` in the details page.
![Screenshot 2024-12-20 at 10 43 11 AM (2)](https://github.com/user-attachments/assets/bea8df24-3d0f-486e-ab69-4a1d6e01543c)

Reply:
Another issue found: When clicking `Restore Latest Backup` for 1217-test (backup target: default), the modal doesn’t appear.

Reply:
Create Diasater Recovery Volime for 1217-test (default) is good for me.
<img width="1425" alt="image" src="https://github.com/user-attachments/assets/401dbeea-e855-4a37-b6f6-554fe468f3c0" />

Reply:
Like the backend bug that the `/v1/backupvolumes/bug-test-a98258ed?action=backupList` (default) return cifs backup data. @mantissahz will fix from backend first.

Reply:
@houhoucoop , @mantissahz , let's review and merge this PR to add update backup target feature first. We can fix this bug in next PR working with backend change.